File: packages/types/src/share.ts
// Secure share types — one-time encrypted share links

export interface CreateShareRequest {
  /** Base64-encoded encrypted content */
  encrypted_data: string;
  /** Max number of views (undefined = unlimited until expiry) */
  max_views?: number;
  /** Expiry duration in hours */
  expires_in_hours: number;
}

export interface ShareResponse {
  id: string;
  share_url: string;
  expires_at: string;
}

export interface ShareContent {
  encrypted_data: string;
  remaining_views?: number;
  expires_at: string;
}

File: crates/vaultic-types/src/share.rs
//! Secure share types — encrypted one-time share links.

use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Debug, Serialize, Deserialize)]
pub struct CreateShareRequest {
    /// Base64-encoded encrypted content
    pub encrypted_data: String,
    /// Max number of views (None = unlimited until expiry)
    pub max_views: Option<i32>,
    /// Expiry duration in hours
    pub ttl_hours: Option<i32>,
    /// Optional link to vault item (requires sync enabled)
    pub vault_item_id: Option<Uuid>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct CreateShareResponse {
    pub share_id: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct ShareContentResponse {
    pub encrypted_data: String,
}
